### PR TITLE
Not found error page

### PR DESCRIPTION
**Pull request by kulmann**
## Description
This PR brings a dedicated `not found` page for the files app. It can be displayed by navigating into a folder and then deleting it in another tab. This is true for a) the `All files` page and b) public links.
The `not found` page will show a link to recover from the page. For `All files` it's offering to go back to the home of the `All files` page and for public links it is offering to reload the root of the public link.

As a bonus, this PR brings a new `options.homeFolder` option for the `config.json` file. It can be used to provide a default path for the `All files` page. The user is not jailed in that path, full breadcrumbs are still usable.

As one more bonus, this PR now makes sure that we also show the breadcrumbs for invalid paths. If you navigate to paths that are still not valid, the same `not found` page will appear, which is a valid and possibly helpful state.

## Related Issue

- Fixes https://github.com/owncloud/phoenix/issues/4248
- Fixes https://github.com/owncloud/ocis/issues/1010

## Motivation and Context
UX improvement

## How Has This Been Tested?

- drone

## Screenshots (if appropriate):
Screenshot 2020-12-03 at 16 11 22

Screenshot 2020-12-03 at 16 11 46

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] fix acceptance tests
- [ ] look into https://www.npmjs.com/package/vue-inline-svg for styling of svgs as inline svg

### CODE REVIEW

File: tests/acceptance/stepDefinitions/generalContext.js [hunk 1]
```diff
@@ -43,9 +43,18 @@ function setconfig(key, subkey, value, configfile) {
     data[key] = {}
   }
   data[key][subkey] = value
+  console.log(data)
```
Comment (LukasHirt) [inline tests/acceptance/stepDefinitions/generalContext.js:46]: Forgotten log?

File: changelog/unreleased/resource-not-found-error-page [hunk 2]
```diff
@@ -0,0 +1,6 @@
+Change: Show dedicated 404 page for invalid resource references
+
+When visiting a public link or the `All files` files list with an invalid resource in the URL (e.g. because it was deleted in the meantime) we now show a dedicated page which explains
+the the resource could not be found and which offers a link to go back to the respective root (»All files« home location or the root of the public link).
```
Comment (LukasHirt) [inline changelog/unreleased/resource-not-found-error-page:4]: `explains the the`

**Review by LukasHirt — APPROVED** [review]
Just two small comments, otherwise amazing! 🚀

File: changelog/unreleased/resource-not-found-error-page (hunk 2 above)
Comment (kulmann) [reply changelog/unreleased/resource-not-found-error-page]: 👍

File: tests/acceptance/stepDefinitions/generalContext.js (hunk 1 above)
Comment (kulmann) [reply tests/acceptance/stepDefinitions/generalContext.js]: yes, thanks 😅

**Pull request merged**